app/services/search_service.rb

```diff
@@ -167,27 +167,26 @@ def tagNearbyNodes(srchString, tagName)
 
   # GET X number of latest people/contributors and package up as a DocResult
   # X = srchString
-  def recentPeople(srchString, tagName = nil)
+  def people_locations(srchString, tagName = nil)
     sresult = DocList.new
 
-    nodes = Node.all.order("changed DESC").limit(srchString).distinct
-    users = []
-    nodes.each do |node|
-      if node.author.status != 0
-        if tagName.blank?
-          users << node.author.user
-        else
-          users << node.author.user if node.author.user.has_tag(tagName)
-        end
-      end
+    user_scope = User.where('rusers.status <> 0')\
+                     .joins(:user_tags)\
+                     .where('value LIKE "lat:%"')\
+                     .joins(:revisions)\
```

Comment:
This might be a bit heavy if all you're wanting is the timestamp. I wonder if you could select just the timestamp, but you might have to do `.select('node_revisions.timestamp', ` plus every other column (Not sure if you can wildcard `*` multiple columns...) so then the way to get past that could be to get the absolute minimum number of columns with a `select()`, collect the `nid`s with `nids = query.collect(&:nid)` and then run a second query, like:

```ruby
User.where(nid: nids)
```

This way the original query is very narrowly restricted, then a second query collects the full User records. This is a bit messy looking but you can see how it might result in a more efficient initial query followed by a narrower second query which gets all the `User` table bits and pieces you're looking for, without the messy joins. Does this make sense?

Reply:
To be honest I don't know if it makes sense! :stuck_out_tongue_closed_eyes: 

Actually, under rails, the query being executed is: `SELECT  DISTINCT "rusers".* FROM "rusers" INNER JOIN "user_tags" ON "user_tags"."uid" = "rusers"."id" INNER JOIN "node" ON "node"."uid" = "rusers"."id" INNER JOIN "node_revisions" ON "node_revisions"."nid" = "node"."nid" WHERE (rusers.status <> 0) AND (value LIKE "lat:%") ORDER BY node_revisions.timestamp DESC LIMIT ?`. We are only selecting the data from the user table, which is what we need at the end of the day, and we use the timestamp just to sort the results. Of course, I do not know exactly how mysql runs this query, but it is unlikely that it actually returns the entire secondary table to only then select what the query explicitly asks for...